packages/api/src/users/users.mapper.ts

```diff
@@ -0,0 +1,32 @@
+import type { User } from '@coderscamp/shared/models/user';
+
+import type { GithubDTO } from '../auth/github/github.model';
+import type { UserDTO } from './users.model';
+
+export class UsersMapper {
+  static toDomain(value: UserDTO): User {
+    return value as User;
```

Comment:
Why do we need to cast type in this and nearly all other methods? It doesn't seem right as we might lose type-safety.

Reply:
This mapper function doesn't have any sense. You are casting and you do not validate/normalize any value.
Please read, https://docs.nestjs.com/techniques/serialization
https://docs.nestjs.com/pipes#class-validator 
You can also use `class-transformer-validator`. 
As example 
```
  static toDomain(
    props: Partial<UserCore>,
    options: { keyCode?: string } = {},
  ): User {
    const createUser = options.keyCode ? User.create : User.recreate;

    return createUser(
      transformAndValidateSync(UserCore, props, {
        transformer: { excludeExtraneousValues: true },
        validator: { validationError: { target: false, value: false } },
      }),
      options.keyCode,
    );
  }

  static toPlain(child: User): UserCore {
    const props = child.getProps();

    return classToPlain(props, { excludeExtraneousValues: true }) as UserCore;
  }```